src/functions/useIntersectionTop.ts

```diff
@@ -0,0 +1,66 @@
+import { onMounted, onUnmounted, ref } from "vue";
+
+const DETECTOR_COUNT = 10;
+const INTERSECTION_ERROR_MARGIN = 1
+
+const detectorContainer = document.createElement("div");
+Object.assign(detectorContainer.style, {
+  position: "absolute",
+  top: "0",
+  left: "0",
+  opacity: "0",
+  pointerEvents: "none"
+});
+const detectors = new Array(DETECTOR_COUNT).fill(0).map(() => {
+  const detector = document.createElement("div");
+  Object.assign(detector.style, {
+    position: "absolute",
+    top: "0",
+    left: "0",
+    width: "100%"
+  });
+  detectorContainer.appendChild(detector);
+  return detector;
+});
+document.body.appendChild(detectorContainer);
+
+export default function useIntersectionTop() {
+  const top = ref(0);
+  let observer: IntersectionObserver;
+
+  onMounted(() => {
+    const scrollHeight = document.documentElement.scrollHeight;
+    const scrollWidth = document.documentElement.scrollWidth;
+    detectorContainer.style.height = scrollHeight + "px";
+    detectorContainer.style.width = scrollWidth + "px";
+    const detectorHeight = Math.ceil(
+      document.documentElement.scrollHeight / DETECTOR_COUNT
+    );
+    const threshold = [...Array(detectorHeight + 1).keys()].map(
+      i => i / detectorHeight
+    );
+    detectors.forEach((detector, index) => {
+      detector.style.top = index * detectorHeight + "px";
+      detector.style.height = detectorHeight + "px";
+    });
+    observer = new IntersectionObserver(
+      ([entry]) => {
```

Comment:
I am going to merge this PR since it works fine. Can you please let me know why we only need to extract the first entry with `([entry])`?

Reply:
Note that the entries passed into the callback only contain those triggered the "Intersection Event". The infomation given by the first entry is sufficient for our case to get the iframe's position, and the reason we have multiple entries is that a single Detector will not trigger the callback in a particular situation.